app/registries/foreman/plugin.rb

```diff
@@ -40,19 +40,31 @@ class PluginRequirementError < Foreman::Exception; end
   class Plugin
     @registered_plugins = {}
     @tests_to_skip = {}
-    @fact_importer_registry = Plugin::FactImporterRegistry.new
-    @report_scanner_registry = Plugin::ReportScannerRegistry.new
-    @report_origin_registry = Plugin::ReportOriginRegistry.new
-    @medium_providers = Plugin::MediumProvidersRegistry.new
-    @graphql_types_registry = Plugin::GraphqlTypesRegistry.new
+    @registries = {}
 
     class << self
       attr_reader   :registered_plugins
-      attr_accessor :tests_to_skip, :report_scanner_registry,
-        :report_origin_registry, :medium_providers,
-        :graphql_types_registry, :fact_importer_registry
+      attr_reader   :registries
+      attr_accessor :tests_to_skip
       private :new
 
+      def initialize_registries(registries = {})
```

Comment:
I wonder if this should be private and called as part of `clear`.

Another method is to define specific methods for these methods. If the registry is not in `@registries`, it's lazily initialized. Then when you call clear, it will be initialized automatically when used.

Reply:
:+1: done